### PR TITLE
refactor update_work and test_update_work to use requests

### LINKED ISSUE #2852: Replace all urllib / urllib2 imports and uses with `requests`
`urllib2` is old, and a bit clunky -- I've found recently it can't do some of things I needed it to do, and it also causes problem for moving to Python3, and generally seems to make the code more complex.

Instead of re-using it, and adding new features that use the older mechanisms, can we consider refactoring to use `requests` instead?

My PR where I stripped out `urllib2` and `StringIO` etc to enable `HEAD` requests:  #2838

a similar PR that went with using `requests` to avoid old bugs: #2779 

By using `requests` (rather than adding imports from `six`) we should be getting a better tool for making requests, and reducing code complexity, and the number of imports. 

Relates to #2308 since a common use of `simplejson` is to parse `urllib2` responses. `requests` has a builtin `json()` method so an extra import is not required.

### PR DESCRIPTION
**Pull request by aasifkhan7**
Closes #2852
related to #3694 

@cclauss Refactor only [`update_work.py`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/solr/update_work.py) and [`test_update_work.py`](https://github.com/internetarchive/openlibrary/blob/master/openlibrary/tests/solr/test_update_work.py)

### CODE REVIEW

File: openlibrary/tests/solr/test_update_work.py [hunk 1]
```diff
@@ -425,8 +427,21 @@ def test_ddcs(self, doc_ddcs, solr_ddcs, sort_ddc_index):
...
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+        
+        def json(self):
+            return self.json_data
+        
+    if args[0] == "http://solr:8080/solr/select?wt=json&json.nl=arrarr&q=author_key:OL25A&sort=edition_count+desc&rows=1&fl=title,subtitle&facet=true&facet.mincount=1&facet.field=subject_facet&facet.field=time_facet&facet.field=person_facet&facet.field=place_facet":
+        return MockResponse({"facet_counts": {"facet_fields": {"place_facet": [], "person_facet": [], "subject_facet": [], "time_facet": []}},"response":{"numFound": 0}}, 200)
```
Comment (cclauss) [inline openlibrary/tests/solr/test_update_work.py:440]:
```suggestion
    if args[0] == (
        "http://solr:8080/solr/select?wt=json&json.nl=arrarr&q=author_key:OL25A&"
        "sort=edition_count+desc&rows=1&fl=title,subtitle&facet=true&facet.mincount=1&"
        "facet.field=subject_facet&facet.field=time_facet&facet.field=person_facet&"
        "facet.field=place_facet"
    ):
        return MockResponse(
            {
                "facet_counts": {
                    "facet_fields": {
                        "place_facet": [],
                        "person_facet": [],
                        "subject_facet": [],
                        "time_facet": [],
                    }
                },
                "response": {"numFound": 0},
            },
            200,
        )
```

Comment (cclauss) [reply openlibrary/tests/solr/test_update_work.py:440]: python3 -m pip install black
black my_code_snippit.py  # will autoformat the code for readability.

**Comment by cclauss** [comment]
Not (yet) mandatory... https://travis-ci.org/github/internetarchive/openlibrary/jobs/717652021#L433-L447

File: openlibrary/tests/solr/test_update_work.py (hunk 1 above)
Comment (aasifkhan7) [reply openlibrary/tests/solr/test_update_work.py:440]: Ohh. Thanks!

Comment (aasifkhan7) [reply openlibrary/tests/solr/test_update_work.py]: Done. @cclauss

File: openlibrary/tests/solr/test_update_work.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import pytest
+import unittest
+import mock
```
Comment (cclauss) [inline openlibrary/tests/solr/test_update_work.py:3]:
```suggestion
try:
    from unittest import mock
except ImportError:
    import mock
```

**Comment by aasifkhan7** [comment]
I also see this in the changed files - @cclauss - not very sure about this, but it shouldn't be present, right? maybe i did `git add .` at some point which committed the submodule changes too. 

Screenshot 2020-08-15 at 9 04 36 AM

**Comment by aasifkhan7** [comment]
@cclauss I just confirmed - this _does_ change the submodule settings for infogami - it checks it out to some other commit, which I had done locally. will try to revert that. 😅

**Comment by aasifkhan7** [comment]
Done @cclauss

**Comment by cclauss** [comment]
@cdrini @mekarpeles @hornc I have approved this PR.  It would be good to get your reviews.

File: openlibrary/solr/update_work.py [hunk 3]
```diff
@@ -42,14 +42,14 @@
...
+def urlopen(url, params=None, data=None):
     version = "%s.%s.%s" % sys.version_info[:3]
     user_agent = 'Mozilla/5.0 (openlibrary; %s) Python/%s' % (__file__, version)
     headers = {
         'User-Agent': user_agent
     }
-    req = urllib.request.Request(url, data, headers)
-    return urllib.request.urlopen(req)
+    response = requests.post(url, params=params, data=data, headers=headers)
+    return response
```
Comment (cdrini) [inline openlibrary/solr/update_work.py:52]:
```suggestion
    return response
```

**Review by cdrini — APPROVED** [review]
Lgtm! I tested `make reindex-solr` works and logs `urlopen` without errors

**Pull request merged**